### PR TITLE
[FEAT] add code of PhysicsRegression model

### PR DESCRIPTION
**Pull request by lkyu-ly**
This example integrates the Paddle version of PhysicsRegression into PaddleCFD for the paper [A Neural Symbolic Model for Space Physics](https://www.nature.com/articles/s42256-025-01126-3).

This example is able to run on both Nvidia GPU and Custom Device(using iluvatar for test). CINN is also supported.

Loss comparison of a training test on a rather small config than default:

image

froward diff:

logits_shape: (7, 10570)
torch_dtype: float32
paddle_dtype: float32
mean_abs_error: 5.4308799236e-07
max_abs_error: 5.7220458984e-06
**mean_rel_error: 1.6366435602e-06**
max_rel_error: 3.4902596381e-03

### CODE REVIEW

**Comment by lkyu-ly** [comment]
To switch to \_SimpleIterator for deterministic cross-framework loss alignment:

1. Comment out the `return paddle.io.DataLoader(...)` block above.
2. Uncomment the `class _SimpleIterator` and the final `return _SimpleIterator(...)` below.
3. If aligning against a PyTorch baseline, the PyTorch side MUST apply the same change  
   (replace its DataLoader with an equivalent \_SimpleIterator) — otherwise batch order and  
   RNG state will diverge between the two frameworks, making per-step loss comparison invalid.

Loss alignment results (first epoch, sampled every 50 steps, PyTorch vs Paddle):  
 Shape: (10,) | dtype: float32 vs float32  
 max_abs_diff: 0.007700 mean_abs_diff: 0.003450 rmse: 0.004042  
 max_rel_diff: 0.043800 mean_rel_diff: 0.019583  
 diff range: [-0.007700, 0.003900]
 
 
1777456682972_d
image

**Comment by zhiminzhang0830** [comment]
> To switch to _SimpleIterator for deterministic cross-framework loss alignment:
> 
> 1. Comment out the `return paddle.io.DataLoader(...)` block above.
> 2. Uncomment the `class _SimpleIterator` and the final `return _SimpleIterator(...)` below.
> 3. If aligning against a PyTorch baseline, the PyTorch side MUST apply the same change
>    (replace its DataLoader with an equivalent _SimpleIterator) — otherwise batch order and
>    RNG state will diverge between the two frameworks, making per-step loss comparison invalid.
> 
> Loss alignment results (first epoch, sampled every 50 steps, PyTorch vs Paddle): Shape: (10,) | dtype: float32 vs float32 max_abs_diff: 0.007700 mean_abs_diff: 0.003450 rmse: 0.004042 max_rel_diff: 0.043800 mean_rel_diff: 0.019583 diff range: [-0.007700, 0.003900]
> 
> 1777456682972_d image

最大误差在7e-3的量级好像也有点大，是否存在其他未对齐的问题

**Comment by zhiminzhang0830** [comment]
image
此案例运行报错，辛苦再查看下。运行命令：bash ./bash/train.sh

**Comment by lkyu-ly** [comment]
> 此案例运行报错，辛苦再查看下。运行命令：bash ./bash/train.sh

可能是本地环境问题，麻烦试用一下aistudio上面创建的项目是否正确运行，https://aistudio.baidu.com/projectdetail/10015658
项目是基于此提交分支的，发布时按readme已测试正常工作。
